tests/conftest.py

```diff
@@ -113,15 +115,21 @@ def fail_test_on_unclosed_tasks(request: FixtureRequest):
     by the end of the test.
     """
 
-    fail_count = request.session.testsfailed
-    loop = asyncio.get_event_loop()
-    loop.set_debug(True)
+    try:
+        fail_count = request.session.testsfailed
+        loop = asyncio.get_running_loop()
+
+        asyncio.set_event_loop(loop)
+        loop.set_debug(True)
 
-    request.addfinalizer(
-        lambda: _error_and_kill_pending_tasks(
-            loop, request.node.name, request.session.testsfailed == fail_count
+        request.addfinalizer(
+            lambda: _error_and_kill_pending_tasks(
+                loop, request.node.name, request.session.testsfailed == fail_count
+            )
         )
-    )
+    except RuntimeError as error:
+        if str(error) != "no running event loop":
+            raise error
```

Comment:
nit: Is it worth having a `pass` explaining that if `str(error) == "no running event loop"` then the test either does not require a loop or will fail when attempting to use the loop?

Reply:
> Is it worth having a `pass`

What do you mean by this?
``` python
except RuntimeError as error:
    if str(error) == "no running event loop":
        pass
    else:
        raise error
```
?

Reply:
Yes exactly that, just a bit more obvious what is intended

Reply:
I would prefer the current `!=` comprehension but I'll add a comment explaining!